<!DOCTYPE html> <!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]--> <!--[if (IE 7)&!(IEMobile)]><html class="no-js lt-ie9 lt-ie8"><![endif]--> <!--[if (IE 8)&!(IEMobile)]><html class="no-js lt-ie9"><![endif]--> <!--[if gt IE 8]><!--> <html class="no-js"><!--<![endif]--> <head> <meta charset="UTF-8"> <meta content="text/html; charset=UTF-8" http-equiv="Content-Type"> <meta http-equiv=X-UA-Compatible content="IE=edge,chrome=1"> <title>Week 3 Walkthrough- Metagenomic Assembly &#8211; Jacob West-Roberts</title> <meta name="description" content="Moon is a minimal, one column jekyll theme for your blog."> <meta name="keywords" content="test"> <!-- Twitter Cards --> <meta name="twitter:card" content="summary"> <meta name="twitter:image" content="/assets/img/nerd.jpg"> <meta name="twitter:title" content="Week 3 Walkthrough- Metagenomic Assembly"> <meta name="twitter:description" content="Metagenomic Assembly!"> <meta name="twitter:site" content="@jwestrob"> <meta name="twitter:creator" content="@jwestrob"> <!-- Open Graph --> <meta property="og:locale" content="en_US"> <meta property="og:type" content="article"> <meta property="og:title" content="Week 3 Walkthrough- Metagenomic Assembly"> <meta property="og:description" content="Metagenomic Assembly!"> <meta property="og:url" content="/Week_3_Walkthrough/"> <meta property="og:site_name" content="Jacob West-Roberts"> <meta property="og:image" content="/assets/img/nerd.jpg"> <link rel="canonical" href="/Week_3_Walkthrough/"> <link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Jacob West-Roberts Feed"> <!-- Handheld --> <meta name="HandheldFriendly" content="True"> <meta name="MobileOptimized" content="320"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <!-- CSS --> <link rel="stylesheet" href="/assets/css/main.css"> <!-- JS --> <script src="/assets/js/modernizr-3.3.1.custom.min.js"></script> <!-- Favicons --> <link rel="apple-touch-icon" href="/assets/img/favicons/apple-icon-precomposed.png"> <link rel="apple-touch-icon" sizes="72x72" href="/assets/img/favicons/apple-icon-72x72.png"> <link rel="apple-touch-icon" sizes="114x114" href="/assets/img/favicons/apple-icon-114x114.png"> <link rel="apple-touch-icon" sizes="144x144" href="/assets/img/favicons/apple-icon-144x144.png"> <link rel="shortcut icon" type="image/png" href="/favicon.png" /> <link rel="shortcut icon" href="/favicon.ico" /> <!-- Background Image --> <style type="text/css">body {background-image:url(/assets/img/pretty_tree_cropped_smaller.jpg); background-repeat: no-repeat; background-size: cover; }</style> <!-- Post Feature Image --> </head> <body> <nav id="dl-menu" class="dl-menuwrapper" role="navigation"> <button class="dl-trigger">Open Menu</button> <ul class="dl-menu"> <li><a href="/">Home</a></li> <li> <a href="#">About</a> <ul class="dl-submenu"> <li> <img src="/assets/img/nerd.jpg" alt="Jacob West-Roberts photo" class="author-photo"> <h4>Jacob West-Roberts</h4> <p>Moon is a minimal, one column jekyll theme for your blog.</p> </li> <li><a href="/about/"><span class="btn btn-inverse">Learn More</span></a></li> <li> <a href="http://twitter.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-twitter-square"></i> Twitter</a> </li> <li> <a href="http://github.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-github"></i> Github</a> </li> </ul><!-- /.dl-submenu --> </li> <li> <a href="#">Posts</a> <ul class="dl-submenu"> <li><a href="/posts/">All Posts</a></li> <li><a href="/tags/">All Tags</a></li> </ul> </li> <li><a href="/projects/" >Projects</a></li> <li><a href="/ESPM_112L/" >ESPM_112L</a></li> </ul><!-- /.dl-menu --> </nav><!-- /.dl-menuwrapper --> <!-- Header --> <header class="header" role="banner"> <div class="wrapper animated fadeIn"> <div class="content"> <div class="post-title "> <h1>Week 3 Walkthrough- Metagenomic Assembly</h1> <h4>03 Feb 2021</h4> <a class="btn zoombtn" href="/projects/"> <i class="fa fa-chevron-left"></i> </a> </div> <h1>Hello and welcome to week 3 of ESPM 112L-</h1> <h1>Metagenomic Data Analysis Lab!</h1> <ul id="markdown-toc"> <li><a href="#assembly-methods" id="markdown-toc-assembly-methods">Assembly methods</a></li> <li><a href="#section-1-assembly" id="markdown-toc-section-1-assembly">Section 1: Assembly</a> <ul> <li><a href="#subsection-assembly-prep" id="markdown-toc-subsection-assembly-prep">Subsection: Assembly prep</a></li> <li><a href="#the-assembly-command" id="markdown-toc-the-assembly-command">The Assembly Command</a></li> </ul> </li> <li><a href="#section-2-assembly-statistics" id="markdown-toc-section-2-assembly-statistics">Section 2: Assembly Statistics</a> <ul> <li><a href="#interpreting-assembly-statistics" id="markdown-toc-interpreting-assembly-statistics">Interpreting assembly statistics</a></li> <li><a href="#getting-assembly-statistics-for-your-sample" id="markdown-toc-getting-assembly-statistics-for-your-sample">Getting assembly statistics for your sample</a></li> </ul> </li> <li><a href="#questions-for-todays-turn-in" id="markdown-toc-questions-for-todays-turn-in">Questions for today’s turn-in:</a></li> </ul> <p>This week we’re going to be looking at metagenome assembly- what it is, how to do it, and best practices.</p> <p>Your samples are enormous (some of the uncompressed .fastq files are &gt;65GB!) so we’re not going to be able to do metagenome assembly on all of these today.</p> <p>What we are going to do is an overview of metagenome assembly- what it is, how to run it, and what software to use.</p> <p>First, <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0169662">here’s a link</a> to a nice paper by Vollmers et al. that describes the most popular software packages for metagenome assembly. It’ll be a nice resource for you in the future if you encounter this again.</p> <p>Second, let’s go over the methods we use in our lab and why we use them.</p> <h1 id="assembly-methods">Assembly methods</h1> <p>Our lab uses <code class="highlighter-rouge">idba_ud</code>, as we tend to get the best results with it (and it has a nice built-in scaffolder). This is by no means a one-size-fits-all solution; different assemblers work to different degrees depending on the type of sample you’re working with, its environment of origin, and your sequencing depth.</p> <p>For a nice example, see the image below from the paper mentioned above:</p> <p>&lt;img src=”https://journals.plos.org/plosone/article/figure/image?size=large&amp;id=info:doi/10.1371/journal.pone.0169662.g002” width=250&gt;</p> <p>As you can see, different assemblers win out over others when looking at particular metrics, but none is consistently better than all the others based on all metrics across different sample types. It’s up to you to decide which one is best for your particular situation, based on the particular traits of each assembler (which are well described in the Vollmers et al. paper above).</p> <p>We can see the effect of using different assemblers on our data, too. I ran <code class="highlighter-rouge">megahit</code> and <code class="highlighter-rouge">metaSPAdes</code> for one of our class samples, <code class="highlighter-rouge">JS_HA1_S133</code>. Here’s what the stats look like in comparison:</p> <hr /> <table> <thead> <tr> <th>Assembly Statistic</th> <th>MEGAHIT</th> <th>metaSPAdes</th> </tr> </thead> <tbody> <tr> <td># Contigs</td> <td>679859</td> <td>2448126</td> </tr> <tr> <td>N50</td> <td>1231 bp</td> <td>578 bp</td> </tr> <tr> <td>Average Sequence Length</td> <td>901.35 bp</td> <td>453.59 bp</td> </tr> <tr> <td>Largest Contig</td> <td>429190</td> <td>502671</td> </tr> </tbody> </table> <hr /> <p>Our lab uses either <code class="highlighter-rouge">metaSPAdes</code>, <code class="highlighter-rouge">idba_ud</code>, or <code class="highlighter-rouge">MEGAHIT</code>; today we’ll be using <code class="highlighter-rouge">metaSPAdes</code>.</p> <hr /> <h1 id="section-1-assembly">Section 1: Assembly</h1> <p>First, we’re going to set up a practice assembly. Navigate to <code class="highlighter-rouge">/class_data/example_assembly</code> and take a look at what’s there. You’ll see two files:</p> <div class="highlighter-rouge"><pre class="highlight"><code>jwestrob@class:/class_data/example_assembly$ ls

JS_WN1_S130_example.trimmed.1.fastq  JS_WN1_S130_example.trimmed.2.fastq
</code></pre></div> <p>Here we have a subset of both the forward and reverse reads from the sample <code class="highlighter-rouge">JS_WN1_S130</code>, which none of the groups have been assigned. Here we have 500,000</p> <h2 id="subsection-assembly-prep">Subsection: Assembly prep</h2> <p>Now we’re going to prepare to run an assembly. Choose your reads, and do the following:</p> <ul> <li><code class="highlighter-rouge">ln -s /class_data/example_assembly/*.fastq ~</code></li> </ul> <p>This will create what’s called a <em>symbolic link</em> in your home directory (your home directory is symbolized by <code class="highlighter-rouge">~</code>) - it’s like copying over a file, but you don’t actually make a new copy. You can just see the filename and operate on it as if you had copied it. If you remove this link, the original will be safe and sound in its original directory.</p> <h2 id="the-assembly-command">The Assembly Command</h2> <p>Now, we’re going to do actual assembly. Remember, <strong>only one person per group should execute one of the following commands! We only have so many compute resources.</strong></p> <div class="highlighter-rouge"><pre class="highlight"><code>#This navigates to your home directory
cd ~
#This displays the help for metaspades
metaspades.py -h
</code></pre></div> <p>Take a look at some of these options.</p> <div class="highlighter-rouge"><pre class="highlight"><code>SPAdes genome assembler v3.15.0 [metaSPAdes mode]

Usage: spades.py [options] -o &lt;output_dir&gt;

Basic options:
  -o &lt;output_dir&gt;             directory to store all the resulting files (required)
  --iontorrent                this flag is required for IonTorrent data
  --test                      runs SPAdes on toy dataset
  -h, --help                  prints this usage message
  -v, --version               prints version

Input data:
  --12 &lt;filename&gt;             file with interlaced forward and reverse paired-end reads
  -1 &lt;filename&gt;               file with forward paired-end reads
  -2 &lt;filename&gt;               file with reverse paired-end reads
  -s &lt;filename&gt;               file with unpaired reads
  --merged &lt;filename&gt;         file with merged forward and reverse paired-end reads
</code></pre></div> <p>So you’re going to need at least two things to run metaSPAdes:</p> <ul> <li>Reads to assemble (your <code class="highlighter-rouge">fastq</code> files)</li> <li>The name of an output directory</li> </ul> <p>Here’s how that command should look:</p> <div class="highlighter-rouge"><pre class="highlight"><code>metaspades.py -t 4 -1 [forward reads] -2 [reverse reads] -o [output_directory name]
</code></pre></div> <p>Let’s run through these options one by one.</p> <ul> <li><code class="highlighter-rouge">-t</code> : The number of threads (separate processes) allotted to the program.</li> <li><code class="highlighter-rouge">-1</code> : The file containing the forward reads.</li> <li><code class="highlighter-rouge">-2</code> : The file containing the reverse reads.</li> <li><code class="highlighter-rouge">-o</code> : The name of the directory to output your assembly to. I suggest <code class="highlighter-rouge">metaspades_out</code>.</li> </ul> <p>With the standard parameters, assembling the example reads with 4 threads took about 2.5 minutes. Similar assemblies take ~4 minutes for <code class="highlighter-rouge">MEGAHIT</code> and ~15 for <code class="highlighter-rouge">idba_ud</code>.</p> <p><em>Please do not run this with more than 4 threads!</em></p> <hr /> <h1 id="section-2-assembly-statistics">Section 2: Assembly Statistics</h1> <p>We’re going to use a script called <code class="highlighter-rouge">contig_stats.pl</code>, written in perl, to generate assembly statistics.</p> <p>Navigate to your metaspades output directory (e.g. <code class="highlighter-rouge">cd metaspades_out</code>), and take a look at the files metaspades generated. Here’s an example output:</p> <div class="highlighter-rouge"><pre class="highlight"><code>jwestrob@class:/class_data/example_assembly/metaspades_out$ ls

assembly_graph_after_simplification.gfa  corrected               K55             run_spades.yaml
assembly_graph.fastg                     dataset.info            misc            scaffolds.fasta
assembly_graph_with_scaffolds.gfa        first_pe_contigs.fasta  params.txt      scaffolds.paths
before_rr.fasta                          input_dataset.yaml      pipeline_state  spades.log
contigs.fasta                            K21                     quast_out       strain_graph.gfa
contigs.paths                            K33                     run_spades.sh   tmp

</code></pre></div> <p>You’ll notice a lot of different unfamiliar files, but two that should hopefully stand out: <code class="highlighter-rouge">contigs.fasta</code> and <code class="highlighter-rouge">scaffolds.fasta</code>. We want <code class="highlighter-rouge">scaffolds.fasta</code> - remember, contigs are stitched together to make scaffolds.</p> <p>Let’s run <code class="highlighter-rouge">contig_stats.pl</code> to get a good idea of how well the assembly ran-</p> <div class="highlighter-rouge"><pre class="highlight"><code>#This command will generate a file called 'scaffolds.fasta.summary.txt'
contig_stats.pl -i scaffolds.fasta -o scaffolds.fasta
</code></pre></div> <p>Let’s look now at the resulting contig stats- try <code class="highlighter-rouge">less -S scaffolds.fasta.summary.text</code>. You’ll see something like this:</p> <div class="highlighter-rouge"><pre class="highlight"><code>Length distribution
===================

Range    	# sequences (%)	# bps (%)
0-100:  	190 (0.24%)	14076 (0.04%)
100-500:  	65811 (83.44%)	17993525 (60.05%)
500-1000:  	10091 (12.79%)	6788583 (22.65%)
1000-5000:  	2694 (3.41%)	4317525 (14.4%)
5000-10000:  	70 (0.08%)	459324 (1.53%)
10000-20000:  	11 (0.01%)	144408 (0.48%)
20000-50000:  	3 (0%)	81549 (0.27%)
50000-:     	2 (0%)	163131 (0.54%)

General Information
==================

Total number of sequences: 78872
Total number of bps:       29962121
Average sequence length:   379.88 bps.
N50:                       364 bps
</code></pre></div> <h2 id="interpreting-assembly-statistics">Interpreting assembly statistics</h2> <p>Important stats to remember are:</p> <ul> <li> <p>N50 (Median contig size; half of contigs are above this size, half below)</p> </li> <li> <p>Average sequence length</p> </li> <li> <p>Total number of sequences (you want fewer, larger contigs!)</p> </li> <li> <p>Number of large (50,000+ bp) contigs</p> </li> </ul> <h2 id="getting-assembly-statistics-for-your-sample">Getting assembly statistics for your sample</h2> <ul> <li> <p>Take a look at the directory for your sample (e.g. <code class="highlighter-rouge">/class_data/assemblies/JS_HA1_S131</code> or similar). <em>If you don’t remember your sample ID ask me for help!</em></p> </li> <li> <p>Using the procedure above, generate contig stats for your scaffold file (<code class="highlighter-rouge">[samplename]_scaffold.fa</code>).</p> </li> </ul> <h1 id="questions-for-todays-turn-in">Questions for today’s turn-in:</h1> <ol> <li> <p>What is the N50 of the contigs (<em>not</em> the scaffolds) from the example assembly? (Go back and repeat the procedure to run <code class="highlighter-rouge">contig_stats.pl</code> on the scaffolds from the example assembly, but choose <code class="highlighter-rouge">contigs.fasta</code> instead of <code class="highlighter-rouge">scaffolds.fasta</code>)</p> </li> <li> <p>What differentiates contigs from scaffolds? Should an assembly yield more scaffolds than contigs, or vice versa?</p> </li> <li> <p>What is the length of the largest contig in your sample’s assembly?</p> </li> </ol> <hr /> <p>You did it!</p> <p><img src="https://i1.wp.com/i.imgflip.com/v80vq.jpg?resize=640%2C359&amp;ssl=1" alt="You get a genome!" /></p> <p>(PC: https://www.molecularecologist.com/2015/12/post-holiday-gift-ideas-a-draft-genome/)</p> <div class="entry-meta"> <br> <hr> <span class="entry-tags"><a href="/tags/#test" title="Pages tagged test" class="tag"><span class="term">test</span></a></span> <span class="social-share"> <a href="https://www.facebook.com/sharer/sharer.php?u=/Week_3_Walkthrough/" title="Share on Facebook" class="tag"> <span class="term"><i class="fa fa-facebook-square"></i> Share</span> </a> <a href="https://twitter.com/intent/tweet?text=/Week_3_Walkthrough/" title="Share on Twitter" class="tag"> <span class="term"><i class="fa fa-twitter-square"></i> Tweet</span> </a> <a href="https://plus.google.com/share?url=/Week_3_Walkthrough/" title="Share on Google+" class="tag"> <span class="term"><i class="fa fa-google-plus-square"></i> +1</span> </a> </span> <div style="clear:both"></div> </div> </div> </div> </header> <!-- JS --> <script src="/assets/js/jquery-1.12.0.min.js"></script> <script src="/assets/js/jquery.dlmenu.min.js"></script> <script src="/assets/js/jquery.goup.min.js"></script> <script src="/assets/js/jquery.magnific-popup.min.js"></script> <script src="/assets/js/jquery.fitvid.min.js"></script> <script src="/assets/js/scripts.js"></script> <!-- MathJax --> <script async src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> </body> </html>
