<!DOCTYPE html> <!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]--> <!--[if (IE 7)&!(IEMobile)]><html class="no-js lt-ie9 lt-ie8"><![endif]--> <!--[if (IE 8)&!(IEMobile)]><html class="no-js lt-ie9"><![endif]--> <!--[if gt IE 8]><!--> <html class="no-js"><!--<![endif]--> <head> <meta charset="UTF-8"> <meta content="text/html; charset=UTF-8" http-equiv="Content-Type"> <meta http-equiv=X-UA-Compatible content="IE=edge,chrome=1"> <title>Week 8 Walkthrough- ORF Prediction and Basic Annotation &#8211; Jacob West-Roberts</title> <meta name="description" content="Moon is a minimal, one column jekyll theme for your blog."> <meta name="keywords" content="test"> <!-- Twitter Cards --> <meta name="twitter:card" content="summary"> <meta name="twitter:image" content="/assets/img/nerd.jpg"> <meta name="twitter:title" content="Week 8 Walkthrough- ORF Prediction and Basic Annotation"> <meta name="twitter:description" content="ORF Prediction!"> <meta name="twitter:site" content="@jwestrob"> <meta name="twitter:creator" content="@jwestrob"> <!-- Open Graph --> <meta property="og:locale" content="en_US"> <meta property="og:type" content="article"> <meta property="og:title" content="Week 8 Walkthrough- ORF Prediction and Basic Annotation"> <meta property="og:description" content="ORF Prediction!"> <meta property="og:url" content="/Week_8_Walkthrough/"> <meta property="og:site_name" content="Jacob West-Roberts"> <meta property="og:image" content="/assets/img/nerd.jpg"> <link rel="canonical" href="/Week_8_Walkthrough/"> <link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Jacob West-Roberts Feed"> <!-- Handheld --> <meta name="HandheldFriendly" content="True"> <meta name="MobileOptimized" content="320"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <!-- CSS --> <link rel="stylesheet" href="/assets/css/main.css"> <!-- JS --> <script src="/assets/js/modernizr-3.3.1.custom.min.js"></script> <!-- Favicons --> <link rel="apple-touch-icon" href="/assets/img/favicons/apple-icon-precomposed.png"> <link rel="apple-touch-icon" sizes="72x72" href="/assets/img/favicons/apple-icon-72x72.png"> <link rel="apple-touch-icon" sizes="114x114" href="/assets/img/favicons/apple-icon-114x114.png"> <link rel="apple-touch-icon" sizes="144x144" href="/assets/img/favicons/apple-icon-144x144.png"> <link rel="shortcut icon" type="image/png" href="/favicon.png" /> <link rel="shortcut icon" href="/favicon.ico" /> <!-- Background Image --> <style type="text/css">body {background-image:url(/assets/img/pretty_tree_cropped_smaller.jpg); background-repeat: no-repeat; background-size: cover; }</style> <!-- Post Feature Image --> </head> <body> <nav id="dl-menu" class="dl-menuwrapper" role="navigation"> <button class="dl-trigger">Open Menu</button> <ul class="dl-menu"> <li><a href="/">Home</a></li> <li> <a href="#">About</a> <ul class="dl-submenu"> <li> <img src="/assets/img/nerd.jpg" alt="Jacob West-Roberts photo" class="author-photo"> <h4>Jacob West-Roberts</h4> <p>Moon is a minimal, one column jekyll theme for your blog.</p> </li> <li><a href="/about/"><span class="btn btn-inverse">Learn More</span></a></li> <li> <a href="http://twitter.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-twitter-square"></i> Twitter</a> </li> <li> <a href="http://github.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-github"></i> Github</a> </li> </ul><!-- /.dl-submenu --> </li> <li> <a href="#">Posts</a> <ul class="dl-submenu"> <li><a href="/posts/">All Posts</a></li> <li><a href="/tags/">All Tags</a></li> </ul> </li> <li><a href="/projects/" >Projects</a></li> <li><a href="/ESPM_112L/" >ESPM_112L</a></li> </ul><!-- /.dl-menu --> </nav><!-- /.dl-menuwrapper --> <!-- Header --> <header class="header" role="banner"> <div class="wrapper animated fadeIn"> <div class="content"> <div class="post-title "> <h1>Week 8 Walkthrough- ORF Prediction and Basic Annotation</h1> <h4>04 Mar 2021</h4> <a class="btn zoombtn" href="/projects/"> <i class="fa fa-chevron-left"></i> </a> </div> <p>``</p> <h1>Hello and welcome to week 8 of ESPM 112L-</h1> <h1>Metagenomic Data Analysis Lab!</h1> <ul id="markdown-toc"> <li><a href="#goals-for-today" id="markdown-toc-goals-for-today">Goals for today:</a> <ul> <li><a href="#tools-to-investigate-proteins-of-interest" id="markdown-toc-tools-to-investigate-proteins-of-interest">Tools to investigate proteins of interest:</a></li> </ul> </li> <li><a href="#predicting-orfs-with-ncbi-orf-finder" id="markdown-toc-predicting-orfs-with-ncbi-orf-finder">Predicting ORFs with NCBI ORF Finder</a> <ul> <li><a href="#selecting-a-dna-sequence-on-which-to-predict-proteins" id="markdown-toc-selecting-a-dna-sequence-on-which-to-predict-proteins">Selecting a DNA sequence on which to predict proteins</a></li> <li><a href="#performing-prediction-on-ncbi-orf-finder" id="markdown-toc-performing-prediction-on-ncbi-orf-finder">Performing prediction on NCBI ORF Finder</a></li> <li><a href="#verifying-proteins-with-blastp" id="markdown-toc-verifying-proteins-with-blastp">Verifying proteins with BLASTp</a></li> </ul> </li> <li><a href="#interpro" id="markdown-toc-interpro">Interpro</a></li> </ul> <p>This week’s lab is going to cover both how proteins are predicted in prokaryotes as well as how to go about learning more about interesting proteins you find in metagenomic data. Today we’re going to focus exclusively on proteins you can find in your bins, since those are more interesting (because you know a bit about which organism they came from).</p> <p>In our lab, we use several popular tools to look at interesting proteins, which each have their own advantages and disadvantages. Let’s talk about them, and what they’re each good at.</p> <h1 id="goals-for-today">Goals for today:</h1> <ul> <li>Predict genes, ORFs using NCBI ORF Finder</li> <li>Learn how to use BLASTp to investigate protein sequences</li> <li>Learn how to use and interpret results from Interpro and HMMscan</li> <li>Start playing with KEGG and investigating the metabolic pathways your proteins are part of</li> </ul> <hr /> <h2 id="tools-to-investigate-proteins-of-interest">Tools to investigate proteins of interest:</h2> <ul> <li>Interproscan (most thorough)</li> </ul> <p><a href="https://www.ebi.ac.uk/interpro/search/sequence/">https://www.ebi.ac.uk/interpro/search/sequence/</a></p> <p>This option is the best if you have a protein that’s really unusual and you want to find out exactly what it is. Interproscan uses a large suite of HMMs (probabilistic models that we won’t go over in detail today) to give you a wealth of information about the protein sequence you provide.</p> <ul> <li>Blastp (alignment-based)</li> </ul> <p><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins">https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins</a></p> <p>BLASTp draws on the strength of the NCBI’s public sequence database, as well as a great list of structural models that help you see the domain-level features of your protein sequence, which can tell you a lot about its function.</p> <ul> <li>HMMscan (HMM-based, very fast)</li> </ul> <p><a href="https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan">https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan</a></p> <p>HMMscan allows you to search against a suite of domain-level HMMs, which can tell you a lot about what your protein does, and how it functions. Its companion program, pHMMer, gives you similar results along with a list of similar sequences from the EMBL-EBI’s public database, although this approach yields many fewer hits than running BLASTp and I would recommend using BLAST instead of pHMMer unless you’re pressed for time. It’s really fast, though, and if you’re doing tons of these searches, as I often am in the course of my research, it can be a real time saver.</p> <hr /> <h1 id="predicting-orfs-with-ncbi-orf-finder">Predicting ORFs with NCBI ORF Finder</h1> <h2 id="selecting-a-dna-sequence-on-which-to-predict-proteins">Selecting a DNA sequence on which to predict proteins</h2> <p>Go ahead and go over to <a href="class.ggkbase.berkeley.edu">class.ggkbase.berkeley.edu</a> and log in. Select one of your organisms, and click on it to get a list of the scaffolds in that bin. Select a relatively large scaffold (more than ~20kbp) and click on it. A good way to do this is to sort the sequences by ‘# features’ and find a scaffold with more than 10 genes.</p> <p>&lt;img src=”/assets/img/get_big_scaffold.png” width=250&gt;</p> <h2 id="performing-prediction-on-ncbi-orf-finder">Performing prediction on NCBI ORF Finder</h2> <p>Click on the link to this contig and download the DNA sequence for this contig. Open the fasta file in a plain text editor; select all (cmd+a on Mac or ctrl+a on Windows/Linux), and copy the sequence. Go to NCBI ORF finder (<a href="https://www.ncbi.nlm.nih.gov/orffinder">https://www.ncbi.nlm.nih.gov/orffinder</a>) and paste the sequence into the Query box.</p> <p>&lt;img src=”/assets/img/paste_in_sequence.png” width=250&gt;</p> <p>You will want to use the standard bacterial genetic code, referred to here as “Bacterial, Archaeal, and Plant Plasmid (11)”. A reasonable minimum ORF length is 300, but feel free to try other cutoffs. Hit the submit button to see your potential ORFs.</p> <p>The results show all of the possible genes in all reading frames. You can click on a gene in the viewer or in the list to get its particular sequence. Note: this is ALL of the possibilities across multiple reading frames, some of the resulting proteins are likely not real proteins.</p> <p>&lt;img src=”/assets/img/orf_finder.png” width=250&gt;</p> <h2 id="verifying-proteins-with-blastp">Verifying proteins with BLASTp</h2> <p>BLASTp is a piece of software that will take a protein sequence and search a database for close hits- think a search engine, but for biological sequence data.</p> <p>Verify that your selected protein is real by clicking on it, like in the image below, scrolling down to the bottom left of the page and selecting “BLAST”. If your results show a bunch of other proteins with high sequence identity and defined function, congratulations! You got a nice protein. Keep working with it. Otherwise, find another one, rinse and repeat. The best candidates will have relatively little overlap with other predicted ORFs. All the standard parameters are just fine, so don’t worry about changing anything once you see the page shown in the image below- just scroll down and click BLAST.</p> <p>&lt;img src=”/assets/img/blastp.png” width=250&gt;</p> <p>How well do the results cover your query? Look at the colored bars in the top box to visualize this. Do you get results in the description box that agree on what this protein might be? Do the results have a functional annotation (some kind of specific protein) or does it simply say “hypothetical protein” or “unknown”?</p> <hr /> <h1 id="interpro">Interpro</h1> <p>Now that you have a good ORF that you can trust is real, go ahead and navigate over to Interproscan (<a href="https://www.ebi.ac.uk/interpro/search/sequence/">https://www.ebi.ac.uk/interpro/search/sequence/</a>). Paste this amino acid sequence in as your query and wait for a little while - interpro takes a bit of time, but the results are really good and trustworthy.</p> <p>You’ll get some cool results from interpro which are really interactive and highly detailed, if you have a real protein. If you have a protein with unknown function or that doesn’t look like any well-characterized proteins, you might not. In that case, just go back to NCBI ORF finder and pick another protein and repeat this whole process. (If you’ve closed the window with NCBI ORF finder or just don’t like it, you can always get these proteins from class.ggkbase pretty easily too.)</p> <p>Below is a run down of the kinds of information interpro will display for you:</p> <hr /> <p>&lt;img src=”/assets/img/F.png” width=250&gt;</p> <p>Protein family: in InterPro a protein family is a group of proteins that share a common evolutionary origin, reflected by their related functions and similarities in sequence or structure. (The inclusion of protein structure is one of the differences between the general search in NCBI, that only considered sequence homology, and this search against InterPro)</p> <p>&lt;img src=”/assets/img/D.png” width=250&gt;</p> <p>Protein domain: distinct functional and/or structural units in a protein. Usually they are responsible for particular functions or interaction, contributing to the overall role of a protein. Domains may exist in a variety of biological contexts, where similar domains can be found in proteins with different functions.</p> <p>&lt;img src=”/assets/img/R.png” width=250&gt;</p> <p>Repeats are typically short amino acid sequences that are repeated within a protein, and may confer binding or structural properties upon it.</p> <p>&lt;img src=”/assets/img/S.png” width=250&gt;</p> <p>Sites: groups of amino acids that confer certain characteristics upon a protein, and may be important for its overall function. Sites are usually rather small (only a few amino acids long). Some types of sites in InterPro are active sites (involved in catalytic activity, binding sites (bind molecules or ions), post-translational modification sites (chemically modified after the protein is translated), and conserved sites (found in specific types of proteins, but whose function is unknown)</p> <div class="entry-meta"> <br> <hr> <span class="entry-tags"><a href="/tags/#test" title="Pages tagged test" class="tag"><span class="term">test</span></a></span> <span class="social-share"> <a href="https://www.facebook.com/sharer/sharer.php?u=/Week_8_Walkthrough/" title="Share on Facebook" class="tag"> <span class="term"><i class="fa fa-facebook-square"></i> Share</span> </a> <a href="https://twitter.com/intent/tweet?text=/Week_8_Walkthrough/" title="Share on Twitter" class="tag"> <span class="term"><i class="fa fa-twitter-square"></i> Tweet</span> </a> <a href="https://plus.google.com/share?url=/Week_8_Walkthrough/" title="Share on Google+" class="tag"> <span class="term"><i class="fa fa-google-plus-square"></i> +1</span> </a> </span> <div style="clear:both"></div> </div> </div> </div> </header> <!-- JS --> <script src="/assets/js/jquery-1.12.0.min.js"></script> <script src="/assets/js/jquery.dlmenu.min.js"></script> <script src="/assets/js/jquery.goup.min.js"></script> <script src="/assets/js/jquery.magnific-popup.min.js"></script> <script src="/assets/js/jquery.fitvid.min.js"></script> <script src="/assets/js/scripts.js"></script> <!-- MathJax --> <script async src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> </body> </html>
