---
layout: post
title:  "Week 5 Walkthrough- Automatic Binning"
date:   2021-02-12
excerpt: "Automatic binning!"
project: "ESPM_112L"
tag:
- test
comments: true
---


``
<h1>Hello and welcome to week 5 of ESPM 112L-</h1>

<h1>Metagenomic Data Analysis Lab!</h1>

* TOC
{:toc}


# Introduction

---

## Automatic binning

Automatic binning (often shortened to autobinning) is basically the same as the process you used last week to separate out genome bins from a metagenomic assembly, just done automatically. It's very convenient- if you have a bunch of samples, it's very time consuming to manually bin them all- but has some caveats, since a human isn't there to proofread and curate the binning result. That's your job!

This week, I've run two sets of binning software for you - Metabat (https://peerj.com/articles/1165/) and MaxBin2 (https://pubmed.ncbi.nlm.nih.gov/26515820/). Your task today is going to be to use the results from these binning algorithms, as well as the results from your manual binning last week, to make a consolidated bin set using DASTool.

## Bin consolidation

Different binning approaches use different features to separate out genomes. Your manual ggkbase binning, for example, used GC content, coverage and taxonomy; most of these autobinners will use coverage and _k_-mer composition (essentially a way to turn a DNA string into a numeric vector for computers to interpret).

<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41564-018-0171-1/MediaObjects/41564_2018_171_Fig1_HTML.jpg?as=webp" width=250>

But not every autobinner is the same- they differ in algorithms and the features they look at. As a result, binners will give results of varying quality on individual datasets. Take a look at how three binners (CONCOCT, MaxBin2, and Metabat2) perform on the same dataset, and then what bins look like after consolidation with DASTool:

<img src="https://www.microbiologyresearch.org/docserver/fulltext/mgen/6/10/mgen000436-f2.gif" width=250>

You can see that the consolidated bins are overall of much higher quality than the bins generated by any individual binning method shown. And that's what we're going to do today!

---

# Your binning results

In the interest of time, and because of computational constraints, I've run two binners (again, MaxBin2 and Metabat) for you.




# Today's Turn-In

1. What is the size of your largest bin?

2. What is the taxonomy of that organism? (BLAST the S3 or L6 ribosomal proteins as described above)

3. Which of the two automatic binners performed better on your sample? (See the `hqBins.pdf` file generated by DASTool) If the answer is not immediately clear, describe the distribution of genome quality you see.
