---
layout: post
title:  "Week 12 Walkthrough- Population Dynamics!"
date:   2021-04-09
excerpt: "iRep!"
project: "ESPM_112L"
tag:
  - test
comments: true
---


``
<h1>Hello and welcome to week 12 of ESPM 112L-</h1>

<h1>Metagenomic Data Analysis Lab!</h1>

* TOC
{:toc}

This week we're going to generate an alignment like you did last week, but we're going to visualize that alignment in geneious so you can see two things:

1. What coverage actually looks like and what it means
2. SNP-level microdiversity in a population genome

Let's get to it!

# First step: Making your alignment

This week, making your alignment is going to be a bit easier than last week, since we're going to use a different aligner, called `bbmap.sh`. Don't worry- even though it's a different tool, it's actually much easier to use, and it will generate your alignment much faster!

First, make a folder in your home directory (call it `Lab12` or whatever you'd prefer.) Then, as we did last week, copy in a bin of your choice (from your sample!) to this directory.

Remember to make sure the genome you choose is of generally good quality- at least 1.5Mbp, and the fewer contigs the better!

And, as with last week, we're going to make symbolic links to your reads into this folder to make them easier to access. Trust me- this saves a lot of typing time and hard drive space.

```
mkdir ~/Lab12
cp /class_data/all_contigs/[bin contigs file].fna ~/Lab12

#Go to that directory
cd ~/Lab12

#Make symbolic links to reads
ln -s /class_data/reads/[SAMPLE NAME]/*.fastq.gz .
```

Awesome- now that you've done that, let's go ahead and run your alignment. You don't have to generate the index in a separate step like you did last week with `bowtie2`, so that saves you a step here!

Importantly, make sure to specify `threads=6` when you run this; otherwise `bbmap.sh` will take all the available threads and make it difficult for everyone to run their alignments!

```
#Remember you're now located in ~/Lab12

bbmap.sh pigz=t unpigz=t threads=6 ref=[bin contigs file].fna in1=[FORWARD READS FILE].fastq.gz in2=[REVERSE READS FILE].fastq.gz out=stdout.sam | shrinksam | sambam > [genome name].shrink.sort.bam
```
Just put in values for `[bin contigs file]`, the forward and reverse reads files, and name the final output (`[genome name].shrink.sort.bam`) what you'd like- I would name it based on the name of the genome you're looking at. Say we're using `JS_HB1_S134_bin_31.fna`; in that case I would name the final output `JS_HB1_S134_bin_31.shrink.sort.bam`.

This process should only take a few minutes. `bbmap.sh` is generally much faster than `bowtie2`. Note that this is, of course, dependent on both the size of the genome you've chosen and the number of reads in your dataset that you're aligning to the reference genome.



The additional stuff at the end of the command is various bits of processing; filtering, sorting, and shrinking your alignment file by removing uninformative bits. Let's talk for a bit about what's happening there.

## Interlude: .sam files, .bam files, samtools, oh my!

There are two kinds of read alignment files, which are generated by read mapping programs like `bowtie2` (which you used last week) and `bbmap`, which you're using this week.

The first is a `.sam` file, which stands for Sequence Alignment Map. They look like this:

![SAM file](https://miro.medium.com/max/988/0*CePvh8XpBcbC1HjZ.png)

The information here basically is:

- Where on the reference genome your reads align
- How well those reads match your reference

Which is fantastically useful information, as you've probably surmised due to our having generated such alignments several times so far in lab. However, there's a caveat: it's all in text format! This ends up being very costly in terms of hard drive space (they can each be many gigabytes in size) and computational power (if you have to load a 30GB file every time you do something, it gets old quick).

In order to reduce the amount of space these files take up, we first shrink the file by removing unmapped sequences from the `.sam` file (with `shrinksam`), then convert it to a much smaller binary file, called a `.bam` (with `sambam`).

A `.bam` file contains the same information as a `.sam` file, but in a much smaller package. This is more efficient for programs to load if you want to analyze your data, and more efficient in terms of how much space it's taking up on your hard drive!

The command as I've written it above actually never even writes a `.sam` file to the hard drive, instead feeding the alignment information directly into programs which will process it and then turn it into a `.bam` file. If you're interested in further particulars of how this command works, let me know, and I'll be happy to explain it to you in greater detail.

# Geneious

Now that you've finished generating this alignment, let's download both the alignment (the `.bam` file) and the bin contigs fasta file you used to generate the alignment to your local PC; we're going to boot them up in geneious.

Geneious is a versatile platform for doing genomics analysis in a GUI (Graphical User Interface)-based environment. It's especially great for viewing alignments between reads and reference fasta files to see coverage patterns and SNP microdiversity, both of which we're going to do today.

If you haven't already, please download geneious and activate the free trial, which will last two weeks (plenty of time for you to use it for your projects, if you wish). <a href="https://www.geneious.com/free-trial/">https://www.geneious.com/free-trial/</a>

Before you do anything else, make sure your `.bam` and `.fasta` files are both downloaded from the cluster!

Once geneious has been activated, go ahead and boot it up; here's how to visualize your alignment in geneious.

1. Import your `.fasta` file first!

Once geneious has started, go up to `File -> Import -> From File` and select your genome fasta. When you see the pop-up below, say 'create sequence list'.

<img src="/assets/img/sequence_list.png" width=250>

2. Now import your `.bam` file.

Go again to `File -> Import -> From File` and select your .bam file. **Make sure that you've clicked on your fasta file to select it before you do so!**

Once you've got it imported, you'll see a bunch of scaffold/contig names. Click on one, and let's see the reads that align to that scaffold. It should look something like this, although all coverage patterns will be different:

<img src="/assets/img/initial_coverage.png" width=250>

Let's take a moment to see what's going on here.

Zoom in (scroll as you would on a browser, or use `ctrl` + `+`) and you  can see something very different:

<img src="/assets/img/zoomy_coverage.png" width=250>

See now how there are highlighted nucleotides? These indicate positions in the reads which **disagree** with your reference genome. (e.g. Your reference genome says GATTACA and the read might say GATAACA; the fourth nucleotide is different between the two.)

Also note that if you roll your cursor over a given read, that read's mate pair will be highlighted.
