---
layout: post
title:  "Week 5 Walkthrough- Automatic Binning"
date:   2021-02-12
excerpt: "Automatic binning!"
project: "ESPM_112L"
tag:
- test
comments: true
---


``
<h1>Hello and welcome to week 5 of ESPM 112L-</h1>

<h1>Metagenomic Data Analysis Lab!</h1>

* TOC
{:toc}


# Introduction

---

## Automatic binning

Automatic binning (often shortened to autobinning) is basically the same as the process you used last week to separate out genome bins from a metagenomic assembly, just done automatically. It's very convenient- if you have a bunch of samples, it's very time consuming to manually bin them all- but has some caveats, since a human isn't there to proofread and curate the binning result. That's your job!

This week, I've run two sets of binning software for you - Metabat (https://peerj.com/articles/1165/) and MaxBin2 (https://pubmed.ncbi.nlm.nih.gov/26515820/). Your task today is going to be to use the results from these binning algorithms, as well as the results from your manual binning last week, to make a consolidated bin set using DASTool.

## Bin consolidation

Different binning approaches use different features to separate out genomes. Your manual ggkbase binning, for example, used GC content, coverage and taxonomy; most of these autobinners will use coverage and _k_-mer composition (essentially a way to turn a DNA string into a numeric vector for computers to interpret).

<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41564-018-0171-1/MediaObjects/41564_2018_171_Fig1_HTML.jpg?as=webp" width=250>

But not every autobinner is the same- they differ in algorithms and the features they look at. As a result, binners will give results of varying quality on individual datasets. Take a look at how three binners (CONCOCT, MaxBin2, and Metabat2) perform on the same dataset, and then what bins look like after consolidation with DASTool:

<img src="https://www.microbiologyresearch.org/docserver/fulltext/mgen/6/10/mgen000436-f2.gif" width=250>

You can see that the consolidated bins are overall of much higher quality than the bins generated by any individual binning method shown. And that's what we're going to do today!

---

# Your binning results

In the interest of time, and because of computational constraints, I've run two binners (again, MaxBin2 and Metabat) for you. DASTool takes as input a file called a `scaffolds2bin` file; this is a file that shows which scaffold belongs to which bin. Each binner has different contig assignments- they make different decisions on which bins the contigs should be placed in- and so we generate a `scaffolds2bin` file for each binner.

In the directory for your sample (`/class_data/assemblies/[YOUR SAMPLE NAME HERE]`) there should be a folder called `binning`. Navigate there, and you'll see four important files: your contigs file, a `scaffolds2bin.tsv` file for `Metabat`, a `scaffolds2bin.tsv` file for `MaxBin2`, and a `scaffolds2bin.txt` file from ggKbase (with the results from your binning last week).

Here's the directory for the example assembly, `JS_HF3_S142`, to demonstrate:

```
[INSERT LS OUTPUT HERE]
```

Now what you need to do is use this information to run `DAS_Tool`.

---

# DAS_Tool

`DAS_Tool` is, like all software you'll use in lab, already installed on the class server. Open the help menu by running `DAS_Tool -h`, and take a look at the options. (Remember, if you're ever running software on the command line and you're confused about how to use it, try running that command with `-h`; almost all the time, it'll show a help menu. Sometimes you need to use `--help` or something similar, but that's down to the individual program.)

## Input

As you can see from the help menu, `DAS_Tool` needs two main inputs: `-i`, a comma-separated list of `scaffolds2bin` files, and `-c`, the contigs file to create your bins from. Here's an example of the list you need to make-

Given you're in a folder containing the following files:

```
MaxBin2.scaffolds2bin.tsv
MetaBat.scaffolds2bin.tsv
ggKbase.scaffolds2bin.tsv
```
You would provide the following as `-i` for `DAS_Tool`:

```
-i MaxBin2.scaffolds2bin.tsv,MetaBat.scaffolds2bin.tsv,ggKbase.scaffolds2bin.tsv
```

And for our contigs file, we provide the path:

```
-c /class_data/assemblies/JS_HF3_S142/JS_HF3_S142_scaffold.fa
```

Your path will be similar. If you have trouble getting the path, remember you can `cd` to the directory containing the fasta file and use `realpath [contigs_file].fa` to get the full path to that file!

If you're in, for example, `/class_data/assemblies/JS_HF3_S142/binning` and you want to access the contigs file, it's one level *above* you, so we need to access it with `..`. In terminal, `.` means your current location (I always pronounce it 'here'), and `..` is the folder above you. So you can say

```
-c ../JS_HF3_S142_scaffold.fa
```
instead of providing the full path.

## Output

But remember, you can't write to folders within the `class_data` folder, so you need to include an output flag that specifies to output in your home directory. Remember, we refer to that with `~`; if you're student20, `~` means `/home/student20`. For me, `~` means `/home/jwestrob`.

First, make a folder called `DAS_Tool` in your home directory, like so:

```
mkdir ~/Das_Tool
```

Then, in your DAS_Tool command, specify it as follows:

```
-o ~/Das_Tool/myOutput
```

(You can call it whatever you want, but include `~/Das_Tool` for sure.)

---

# Interpreting DAS Tool

In that output directory, `~/Das_Tool`, you're going to see a bunch of files, but only two are important for your purposes. Here's an example of what you'll see:

```
LC_0.1_DAS_DASTool_hqBins.pdf              LC_0.1_DAS_proteins.faa
LC_0.1_DAS_DASTool.log                     LC_0.1_DAS_proteins.faa.archaea.scg
LC_0.1_DAS_DASTool_scaffolds2bin.txt       LC_0.1_DAS_proteins.faa.bacteria.scg
LC_0.1_DAS_DASTool_scores.pdf              LC_0.1_DAS.seqlength
LC_0.1_DAS_DASTool_summary.txt             LC_0.1_DAS_vamb.scaffolds2bin.tsv.eval
LC_0.1_DAS_maxbin.scaffolds2bin.tsv.eval   
LC_0.1_DAS_metabat.scaffolds2bin.tsv.eval
```

You want the files ending in `DASTool_scores.pdf`, `DASTool_hqBins.pdf`, and `DASTool_scaffolds2bin.txt`. We're going to use the first to examine how well your binners worked, and the second to upload the new bins to ggKbase.

Download those files (using cyberduck or your favorite alternative), and open up the `DASTool_hqBins.pdf` file to take a look. You'll see something like this:

<img src="/assets/img/dastool_hqbins.png" width=250>

which shows the number of bins each binner generated, as well as how complete these genomes are estimated to be.

Now take a look at the file ending in `DASTool_scores.pdf`, and you'll see something like this:

<img src="/assets/img/dastool_scores.png" width=250>

Notice how DASTool tends to consolidate and eliminate the lower-quality bins, and has a much higher quality score cutoff than the other binners. Most binning software doesn't even take completeness into account, which is why you tend to see binning results that yield numerous low-quality bins.

Now let's take your shiny new set of bins and upload them to ggKbase.

---

# Uploading your bins to ggKbase

Go to class.ggkbase.berkeley.edu and go ahead and log in. Head over to your project page and select 'View Organisms', as you did last week. Up at the top right corner, you'll see a blue wrench icon that says 'Batch Rebinning'; click on it and select 'Rebin File'.

<img src="/assets/img/rebin_file.png" width=250>

Now, select 'Add file', upload that file, and press 'Upload and Rebin'. Wait a moment, and all your new DASTool bins will be ready for you to peruse!

---

# Today's Turn-In

1. What is the highest coverage bin in your sample?

2. What is the taxonomy of that organism?

3. Which of the two automatic binners performed better on your sample? (See the `hqBins.pdf` file generated by DASTool) If the answer is not immediately clear, describe the distribution of genome quality you see.
