<!DOCTYPE html> <!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]--> <!--[if (IE 7)&!(IEMobile)]><html class="no-js lt-ie9 lt-ie8"><![endif]--> <!--[if (IE 8)&!(IEMobile)]><html class="no-js lt-ie9"><![endif]--> <!--[if gt IE 8]><!--> <html class="no-js"><!--<![endif]--> <head> <meta charset="UTF-8"> <meta content="text/html; charset=UTF-8" http-equiv="Content-Type"> <meta http-equiv=X-UA-Compatible content="IE=edge,chrome=1"> <title>Week 5 Walkthrough- Automatic Binning &#8211; Jacob West-Roberts</title> <meta name="description" content="Moon is a minimal, one column jekyll theme for your blog."> <meta name="keywords" content="test"> <!-- Twitter Cards --> <meta name="twitter:card" content="summary"> <meta name="twitter:image" content="/assets/img/nerd.jpg"> <meta name="twitter:title" content="Week 5 Walkthrough- Automatic Binning"> <meta name="twitter:description" content="Automatic binning!"> <meta name="twitter:site" content="@jwestrob"> <meta name="twitter:creator" content="@jwestrob"> <!-- Open Graph --> <meta property="og:locale" content="en_US"> <meta property="og:type" content="article"> <meta property="og:title" content="Week 5 Walkthrough- Automatic Binning"> <meta property="og:description" content="Automatic binning!"> <meta property="og:url" content="/Week_5_Walkthrough/"> <meta property="og:site_name" content="Jacob West-Roberts"> <meta property="og:image" content="/assets/img/nerd.jpg"> <link rel="canonical" href="/Week_5_Walkthrough/"> <link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Jacob West-Roberts Feed"> <!-- Handheld --> <meta name="HandheldFriendly" content="True"> <meta name="MobileOptimized" content="320"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <!-- CSS --> <link rel="stylesheet" href="/assets/css/main.css"> <!-- JS --> <script src="/assets/js/modernizr-3.3.1.custom.min.js"></script> <!-- Favicons --> <link rel="apple-touch-icon" href="/assets/img/favicons/apple-icon-precomposed.png"> <link rel="apple-touch-icon" sizes="72x72" href="/assets/img/favicons/apple-icon-72x72.png"> <link rel="apple-touch-icon" sizes="114x114" href="/assets/img/favicons/apple-icon-114x114.png"> <link rel="apple-touch-icon" sizes="144x144" href="/assets/img/favicons/apple-icon-144x144.png"> <link rel="shortcut icon" type="image/png" href="/favicon.png" /> <link rel="shortcut icon" href="/favicon.ico" /> <!-- Background Image --> <style type="text/css">body {background-image:url(/assets/img/pretty_tree_cropped_smaller.jpg); background-repeat: no-repeat; background-size: cover; }</style> <!-- Post Feature Image --> </head> <body> <nav id="dl-menu" class="dl-menuwrapper" role="navigation"> <button class="dl-trigger">Open Menu</button> <ul class="dl-menu"> <li><a href="/">Home</a></li> <li> <a href="#">About</a> <ul class="dl-submenu"> <li> <img src="/assets/img/nerd.jpg" alt="Jacob West-Roberts photo" class="author-photo"> <h4>Jacob West-Roberts</h4> <p>Moon is a minimal, one column jekyll theme for your blog.</p> </li> <li><a href="/about/"><span class="btn btn-inverse">Learn More</span></a></li> <li> <a href="http://twitter.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-twitter-square"></i> Twitter</a> </li> <li> <a href="http://github.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-github"></i> Github</a> </li> </ul><!-- /.dl-submenu --> </li> <li> <a href="#">Posts</a> <ul class="dl-submenu"> <li><a href="/posts/">All Posts</a></li> <li><a href="/tags/">All Tags</a></li> </ul> </li> <li><a href="/projects/" >Projects</a></li> <li><a href="/ESPM_112L/" >ESPM_112L</a></li> </ul><!-- /.dl-menu --> </nav><!-- /.dl-menuwrapper --> <!-- Header --> <header class="header" role="banner"> <div class="wrapper animated fadeIn"> <div class="content"> <div class="post-title "> <h1>Week 5 Walkthrough- Automatic Binning</h1> <h4>12 Feb 2021</h4> <a class="btn zoombtn" href="/projects/"> <i class="fa fa-chevron-left"></i> </a> </div> <p>``</p> <h1>Hello and welcome to week 5 of ESPM 112L-</h1> <h1>Metagenomic Data Analysis Lab!</h1> <ul id="markdown-toc"> <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a> <ul> <li><a href="#automatic-binning" id="markdown-toc-automatic-binning">Automatic binning</a></li> <li><a href="#bin-consolidation" id="markdown-toc-bin-consolidation">Bin consolidation</a></li> </ul> </li> <li><a href="#your-binning-results" id="markdown-toc-your-binning-results">Your binning results</a></li> <li><a href="#das_tool" id="markdown-toc-das_tool">DAS_Tool</a> <ul> <li><a href="#making-an-output-directory" id="markdown-toc-making-an-output-directory">Making an output directory</a></li> <li><a href="#input" id="markdown-toc-input">Input</a> <ul> <li><a href="#important-explainer" id="markdown-toc-important-explainer">Important: Explainer</a></li> </ul> </li> <li><a href="#input-1" id="markdown-toc-input-1">Input</a></li> <li><a href="#output" id="markdown-toc-output">Output</a></li> <li><a href="#the-final-dastool-command" id="markdown-toc-the-final-dastool-command">The Final Dastool Command</a></li> </ul> </li> <li><a href="#interpreting-das-tool" id="markdown-toc-interpreting-das-tool">Interpreting DAS Tool</a></li> <li><a href="#uploading-your-bins-to-ggkbase" id="markdown-toc-uploading-your-bins-to-ggkbase">Uploading your bins to ggKbase</a></li> <li><a href="#todays-turn-in" id="markdown-toc-todays-turn-in">Today’s Turn-In</a></li> </ul> <h1 id="introduction">Introduction</h1> <hr /> <h2 id="automatic-binning">Automatic binning</h2> <p>The objective of automatic binning (often shortened to autobinning) is the same as the process you used last week to separate out genome bins from a metagenomic assembly, just done computationally rather than manually. It’s very convenient- if you have a bunch of samples. Manual binning is very time consuming and sometimes not effective. Automatic binning has some caveats, though, since a human isn’t there to proofread and curate the binning result. That’s your job!</p> <p>This week, I’ve run one automatic binning program for you - Metabat (https://peerj.com/articles/1165/) and MaxBin2 (https://pubmed.ncbi.nlm.nih.gov/26515820/). Your task today is going to be to use the results from these binning algorithms, as well as the results from your manual binning last week, to make a consolidated bin set using DASTool.</p> <h2 id="bin-consolidation">Bin consolidation</h2> <p>Different binning approaches use different features to separate out genomes. Your manual ggkbase binning, for example, used GC content, coverage and taxonomy; most of these autobinners will use coverage and <em>k</em>-mer composition (essentially a way to turn a DNA string into a numeric vector for computers to interpret).</p> <p>&lt;img src=”https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41564-018-0171-1/MediaObjects/41564_2018_171_Fig1_HTML.jpg?as=webp” width=250&gt;</p> <p>But not every autobinner is the same- they differ in algorithms and the features they look at. As a result, binners will give results of varying quality on individual datasets. Take a look at how three binners (CONCOCT, MaxBin2, and Metabat2) perform on the same dataset, and then what bins look like after consolidation with DASTool:</p> <p>&lt;img src=”https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41564-018-0171-1/MediaObjects/41564_2018_171_Fig2_HTML.jpg?as=webp” width=250&gt;</p> <p>You can see that the consolidated bins are overall of much higher quality than the bins generated by any individual binning method shown. And that’s what we’re going to do today!</p> <hr /> <h1 id="your-binning-results">Your binning results</h1> <p>In the interest of time, and because of computational constraints, I’ve run two binners (again, MaxBin2 and Metabat) for you. DASTool takes as input a file called a <code class="highlighter-rouge">scaffolds2bin</code> file; this is a file that shows which scaffold belongs to which bin. Each binner has different contig assignments- they make different decisions on which bins the contigs should be placed in- and so we generate a <code class="highlighter-rouge">scaffolds2bin</code> file for each binner.</p> <p>In the directory for your sample (<code class="highlighter-rouge">/class_data/assemblies/[YOUR SAMPLE NAME HERE]</code>) there should be a folder called <code class="highlighter-rouge">binning</code>. Navigate there, and you’ll see four important files: your contigs file, a <code class="highlighter-rouge">scaffolds2bin.tsv</code> file for <code class="highlighter-rouge">Metabat</code>, a <code class="highlighter-rouge">scaffolds2bin.tsv</code> file for <code class="highlighter-rouge">MaxBin2</code>, and a <code class="highlighter-rouge">scaffolds2bin.txt</code> file from ggKbase (with the results from your binning last week).</p> <p>Here’s the directory for the example assembly, <code class="highlighter-rouge">JS_HF3_S142</code>, to demonstrate:</p> <div class="highlighter-rouge"><pre class="highlight"><code>[INSERT LS OUTPUT HERE]
</code></pre></div> <p>Now what you need to do is use this information to run <code class="highlighter-rouge">DAS_Tool</code>.</p> <hr /> <h1 id="das_tool">DAS_Tool</h1> <p><code class="highlighter-rouge">DAS_Tool</code> is, like all software you’ll use in lab, already installed on the class server. Open the help menu by running <code class="highlighter-rouge">DAS_Tool -h</code>, and take a look at the options. (Remember, if you’re ever running software on the command line and you’re confused about how to use it, try running that command with <code class="highlighter-rouge">-h</code>; almost all the time, it’ll show a help menu. Sometimes you need to use <code class="highlighter-rouge">--help</code> or something similar, but that’s down to the individual program.)</p> <h2 id="making-an-output-directory">Making an output directory</h2> <p>But remember, you can’t write to folders within the <code class="highlighter-rouge">class_data</code> folder, so you need to include an output flag that specifies to output in your home directory. Remember, we refer to that with <code class="highlighter-rouge">~</code>; if you’re student20, <code class="highlighter-rouge">~</code> means <code class="highlighter-rouge">/home/student20</code>. For me, <code class="highlighter-rouge">~</code> means <code class="highlighter-rouge">/home/jwestrob</code>.</p> <p>First, make a folder called <code class="highlighter-rouge">DAS_Tool</code> in your home directory, like so:</p> <div class="highlighter-rouge"><pre class="highlight"><code>mkdir ~/Das_Tool
</code></pre></div> <h2 id="input">Input</h2> <hr /> <h3 id="important-explainer">Important: Explainer</h3> <p>The following subsections show how to structure individual <em>pieces</em> of the DAS_Tool command. Scroll down to the section labeled “The Final Dastool Command” to see how they’re strung together.</p> <hr /> <p>As you can see from the help menu, <code class="highlighter-rouge">DAS_Tool</code> needs two main inputs: <code class="highlighter-rouge">-i</code>, a comma-separated list of <code class="highlighter-rouge">scaffolds2bin</code> files, and <code class="highlighter-rouge">-c</code>, the contigs file to create your bins from. Here’s an example of the list you need to make-</p> <p>Navigate (cd) to your sample directory (<code class="highlighter-rouge">/class_data/assemblies/[sample_id]</code>) which will contain the following files:</p> <div class="highlighter-rouge"><pre class="highlight"><code>MetaBat.scaffolds2bin.tsv
ggKbase.scaffolds2bin.tsv
</code></pre></div> <p>First thing we’re going to do is copy these over to your directory. Try the following commands:</p> <div class="highlighter-rouge"><pre class="highlight"><code>#Navigate to your sample directory
cd /class_data/assemblies/[sample_id]

#Make a folder in your home directory to put the files in
mkdir ~/DAS_Tool

#Copy the right files to your new directory
cp *scaffold_min1000.fa *.tsv ~/DAS_Tool

#Navigate to that folder
cd ~/DAS_Tool
</code></pre></div> <p>Great! Now that you have all your files set up, let’s go take a look at all the individual parts of the command.</p> <hr /> <h2 id="input-1">Input</h2> <p>Your new directory (<code class="highlighter-rouge">~/DAS_Tool</code>) should look something like this:</p> <div class="highlighter-rouge"><pre class="highlight"><code>JS_HA1_S131_scaffold_min1000.fa   maxbin.scaffolds2bin.tsv
JS_HA1_S131.scaffolds_to_bin.tsv  metabat.scaffolds2bin.tsv
</code></pre></div> <p>You have one fasta format file here (<code class="highlighter-rouge">JS_HA1_S131_scaffold_min1000.fa</code>) containing your DNA from your assembly, and three <code class="highlighter-rouge">scaffolds2bin.tsv</code> files containing the information on which scaffolds belong to which bins.</p> <p>The fasta file you will provide to DAS_Tool with the <code class="highlighter-rouge">-c</code> flag, and the scaffolds2bin files you will provide together, as a comma-separated list, with the <code class="highlighter-rouge">-i</code> flag.</p> <p>Now, given these three <code class="highlighter-rouge">scaffolds2bin.tsv</code> files, you would provide the following as <code class="highlighter-rouge">-i</code> for <code class="highlighter-rouge">DAS_Tool</code>:</p> <div class="highlighter-rouge"><pre class="highlight"><code>-i metabat.scaffolds2bin.tsv,ggKbase.scaffolds_to_bin.tsv
</code></pre></div> <p>And for our contigs file, we provide the path:</p> <div class="highlighter-rouge"><pre class="highlight"><code>-c JS_HF3_S142_scaffold.fa
</code></pre></div> <p>Remember, you should have copied this fasta file (as well as the scaffolds2bin files) over to a folder in your home directory <code class="highlighter-rouge">~/DAS_Tool</code>, which is where you should be running the command. If you get issues saying that DAS_Tool can’t find your scaffolds file, try using <code class="highlighter-rouge">ls</code> to make sure you’re in the same directory as that file, and that it’s spelled correctly in your command!</p> <h2 id="output">Output</h2> <p>You should be running this in a folder in your home directory (e.g. <code class="highlighter-rouge">~/DAS_Tool</code> or similar). Make sure you’ve navigated to that directory with <code class="highlighter-rouge">cd</code> before running. Now specify the prefix of your output. All the files DAS_Tool makes will start with this prefix; name it whatever you want, just don’t name it something that will confuse you later!</p> <div class="highlighter-rouge"><pre class="highlight"><code>-o DAS_Tool
</code></pre></div> <hr /> <h2 id="the-final-dastool-command">The Final Dastool Command</h2> <div class="highlighter-rouge"><pre class="highlight"><code>cd /class_data/assemblies/JS_HF3_S142/

#Remember to make a new directory to run DAS_Tool
mkdir ~/DAS_Tool
cp *scaffold_min1000.fa *.tsv ~/DAS_Tool

#Navigate there and run the command
cd ~/DAS_Tool
DAS_Tool -i maxbin2.scaffolds2bin.tsv,metabat.scaffolds2bin.tsv,JS_HF3_S142_scaffolds2bin.tsv -c JS_HF3_S142_scaffold_min1000.fa -o DAS_Tool
</code></pre></div> <hr /> <h1 id="interpreting-das-tool">Interpreting DAS Tool</h1> <p>In that output directory, <code class="highlighter-rouge">~/Das_Tool</code>, you’re going to see a bunch of files, but only two are important for your purposes. Here’s an example of what you’ll see:</p> <div class="highlighter-rouge"><pre class="highlight"><code>LC_0.1_DAS_DASTool_hqBins.pdf              LC_0.1_DAS_proteins.faa
LC_0.1_DAS_DASTool.log                     LC_0.1_DAS_proteins.faa.archaea.scg
LC_0.1_DAS_DASTool_scaffolds2bin.txt       LC_0.1_DAS_proteins.faa.bacteria.scg
LC_0.1_DAS_DASTool_scores.pdf              LC_0.1_DAS.seqlength
LC_0.1_DAS_DASTool_summary.txt             LC_0.1_DAS_vamb.scaffolds2bin.tsv.eval
LC_0.1_DAS_metabat.scaffolds2bin.tsv.eval
</code></pre></div> <p>You want the files ending in <code class="highlighter-rouge">DASTool_scores.pdf</code>, <code class="highlighter-rouge">DASTool_hqBins.pdf</code>, and <code class="highlighter-rouge">DASTool_scaffolds2bin.txt</code>. We’re going to use the first to examine how well your binners worked, and the second to upload the new bins to ggKbase.</p> <p>Download those files (using cyberduck or your favorite alternative), and open up the <code class="highlighter-rouge">DASTool_hqBins.pdf</code> file to take a look. You’ll see something like this:</p> <p>&lt;img src=”/assets/img/dastool_hqbins.png” width=500&gt;</p> <p>which shows the number of bins each binner generated, as well as how complete these genomes are estimated to be.</p> <p>Now take a look at the file ending in <code class="highlighter-rouge">DASTool_scores.pdf</code>, and you’ll see something like this:</p> <p>&lt;img src=”/assets/img/dastool_scores.png” width=250&gt;</p> <p>Notice how DASTool tends to consolidate and eliminate the lower-quality bins, and has a much higher quality score cutoff than the other binners. Most binning software doesn’t even take completeness into account, which is why you tend to see binning results that yield numerous low-quality bins.</p> <p>Now let’s take your shiny new set of bins and upload them to ggKbase.</p> <hr /> <h1 id="uploading-your-bins-to-ggkbase">Uploading your bins to ggKbase</h1> <p>Go to class.ggkbase.berkeley.edu and go ahead and log in. Head over to your project page and select ‘View Organisms’, as you did last week. Up at the top right corner, you’ll see a blue wrench icon that says ‘Batch Rebinning’; click on it and select ‘Rebin File’.</p> <p>&lt;img src=”/assets/img/rebin_file.png” width=250&gt;</p> <p>Now, select ‘Add file’, upload that file, and press ‘Upload and Rebin’. Wait a moment, and all your new DASTool bins will be ready for you to peruse!</p> <hr /> <h1 id="todays-turn-in">Today’s Turn-In</h1> <ol> <li> <p>What is the highest coverage bin in your sample?</p> </li> <li> <p>What is the taxonomy of that organism?</p> </li> <li> <p>How do the genomes generated by manual binning on ggkbase compare to the automatically generated bins in terms of quality? How about the DAS_Tool generated bins?</p> </li> </ol> <div class="entry-meta"> <br> <hr> <span class="entry-tags"><a href="/tags/#test" title="Pages tagged test" class="tag"><span class="term">test</span></a></span> <span class="social-share"> <a href="https://www.facebook.com/sharer/sharer.php?u=/Week_5_Walkthrough/" title="Share on Facebook" class="tag"> <span class="term"><i class="fa fa-facebook-square"></i> Share</span> </a> <a href="https://twitter.com/intent/tweet?text=/Week_5_Walkthrough/" title="Share on Twitter" class="tag"> <span class="term"><i class="fa fa-twitter-square"></i> Tweet</span> </a> <a href="https://plus.google.com/share?url=/Week_5_Walkthrough/" title="Share on Google+" class="tag"> <span class="term"><i class="fa fa-google-plus-square"></i> +1</span> </a> </span> <div style="clear:both"></div> </div> </div> </div> </header> <!-- JS --> <script src="/assets/js/jquery-1.12.0.min.js"></script> <script src="/assets/js/jquery.dlmenu.min.js"></script> <script src="/assets/js/jquery.goup.min.js"></script> <script src="/assets/js/jquery.magnific-popup.min.js"></script> <script src="/assets/js/jquery.fitvid.min.js"></script> <script src="/assets/js/scripts.js"></script> <!-- MathJax --> <script async src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> </body> </html>
