<!DOCTYPE html> <!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]--> <!--[if (IE 7)&!(IEMobile)]><html class="no-js lt-ie9 lt-ie8"><![endif]--> <!--[if (IE 8)&!(IEMobile)]><html class="no-js lt-ie9"><![endif]--> <!--[if gt IE 8]><!--> <html class="no-js"><!--<![endif]--> <head> <meta charset="UTF-8"> <meta content="text/html; charset=UTF-8" http-equiv="Content-Type"> <meta http-equiv=X-UA-Compatible content="IE=edge,chrome=1"> <title>Week 2 Walkthrough- Read Quality and Trimming &#8211; Jacob West-Roberts</title> <meta name="description" content="Moon is a minimal, one column jekyll theme for your blog."> <meta name="keywords" content="test"> <!-- Twitter Cards --> <meta name="twitter:card" content="summary"> <meta name="twitter:image" content="/assets/img/nerd.jpg"> <meta name="twitter:title" content="Week 2 Walkthrough- Read Quality and Trimming"> <meta name="twitter:description" content="Read Trimming and Quality Control!"> <meta name="twitter:site" content="@jwestrob"> <meta name="twitter:creator" content="@jwestrob"> <!-- Open Graph --> <meta property="og:locale" content="en_US"> <meta property="og:type" content="article"> <meta property="og:title" content="Week 2 Walkthrough- Read Quality and Trimming"> <meta property="og:description" content="Read Trimming and Quality Control!"> <meta property="og:url" content="/Week_2_walkthrough/"> <meta property="og:site_name" content="Jacob West-Roberts"> <meta property="og:image" content="/assets/img/nerd.jpg"> <link rel="canonical" href="/Week_2_walkthrough/"> <link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Jacob West-Roberts Feed"> <!-- Handheld --> <meta name="HandheldFriendly" content="True"> <meta name="MobileOptimized" content="320"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <!-- CSS --> <link rel="stylesheet" href="/assets/css/main.css"> <!-- JS --> <script src="/assets/js/modernizr-3.3.1.custom.min.js"></script> <!-- Favicons --> <link rel="apple-touch-icon" href="/assets/img/favicons/apple-icon-precomposed.png"> <link rel="apple-touch-icon" sizes="72x72" href="/assets/img/favicons/apple-icon-72x72.png"> <link rel="apple-touch-icon" sizes="114x114" href="/assets/img/favicons/apple-icon-114x114.png"> <link rel="apple-touch-icon" sizes="144x144" href="/assets/img/favicons/apple-icon-144x144.png"> <link rel="shortcut icon" type="image/png" href="/favicon.png" /> <link rel="shortcut icon" href="/favicon.ico" /> <!-- Background Image --> <style type="text/css">body {background-image:url(/assets/img/pretty_tree_cropped_smaller.jpg); background-repeat: no-repeat; background-size: cover; }</style> <!-- Post Feature Image --> </head> <body> <nav id="dl-menu" class="dl-menuwrapper" role="navigation"> <button class="dl-trigger">Open Menu</button> <ul class="dl-menu"> <li><a href="/">Home</a></li> <li> <a href="#">About</a> <ul class="dl-submenu"> <li> <img src="/assets/img/nerd.jpg" alt="Jacob West-Roberts photo" class="author-photo"> <h4>Jacob West-Roberts</h4> <p>Moon is a minimal, one column jekyll theme for your blog.</p> </li> <li><a href="/about/"><span class="btn btn-inverse">Learn More</span></a></li> <li> <a href="http://twitter.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-twitter-square"></i> Twitter</a> </li> <li> <a href="http://github.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-github"></i> Github</a> </li> </ul><!-- /.dl-submenu --> </li> <li> <a href="#">Posts</a> <ul class="dl-submenu"> <li><a href="/posts/">All Posts</a></li> <li><a href="/tags/">All Tags</a></li> </ul> </li> <li><a href="/projects/" >Projects</a></li> <li><a href="/ESPM_112L/" >ESPM_112L</a></li> </ul><!-- /.dl-menu --> </nav><!-- /.dl-menuwrapper --> <!-- Header --> <header class="header" role="banner"> <div class="wrapper animated fadeIn"> <div class="content"> <div class="post-title "> <h1>Week 2 Walkthrough- Read Quality and Trimming</h1> <h4>29 Feb 2020</h4> <a class="btn zoombtn" href="/projects/"> <i class="fa fa-chevron-left"></i> </a> </div> <h1>Hello and welcome to week 2 of ESPM 112L-</h1> <h1>Metagenomic Data Analysis Lab!</h1> <ul id="markdown-toc"> <li><a href="#tasks-for-today" id="markdown-toc-tasks-for-today">Tasks for Today</a></li> <li><a href="#1-getting-connected-once-again" id="markdown-toc-1-getting-connected-once-again">1. Getting Connected Once Again</a></li> <li><a href="#2-getting-your-reads" id="markdown-toc-2-getting-your-reads">2. Getting your reads</a></li> <li><a href="#running-fastqc" id="markdown-toc-running-fastqc">Running FastQC</a></li> <li><a href="#how-to-download-files-from-the-cluster-important" id="markdown-toc-how-to-download-files-from-the-cluster-important">How to Download Files From The Cluster (IMPORTANT)</a></li> <li><a href="#trimming-with-sickle" id="markdown-toc-trimming-with-sickle">Trimming with Sickle</a></li> <li><a href="#final-step-fastqc-on-the-trimmed-reads" id="markdown-toc-final-step-fastqc-on-the-trimmed-reads">Final step: FastQC on the trimmed reads</a></li> <li><a href="#questions" id="markdown-toc-questions">Questions</a></li> </ul> <h3 id="tasks-for-today">Tasks for Today</h3> <p>Lab today will hopefully be easy on the technical front, and more about interpreting raw sequence data (what you receive back from a sequencing facility after sending them your extracted DNA).</p> <p>To get your points for the day, please provide the following:</p> <ul> <li>The HTML output of FastQC run before and after trimming</li> <li>Answers to questions (located at the end of the walkthrough)</li> </ul> <h3 id="1-getting-connected-once-again">1. Getting Connected Once Again</h3> <p>Time to get back onto the class server! Working locally won’t work for today’s lab since we need to use data that’s stored on the class server.</p> <p>Get your student ID number and password out, and connect to the cluster however you did last week:</p> <p><code class="highlighter-rouge">ssh studentX@class.ggkbase.berkeley.edu</code></p> <p>Each group will be assigned a sample today, and starting this week you’ll work with that sample. These samples are big, though, and so we’re going to have to take a couple precautions to avoid copying them and using up a ton of disk space!</p> <h3 id="2-getting-your-reads">2. Getting your reads</h3> <p>Make a folder in your home directory (use the <code class="highlighter-rouge">mkdir</code> command) called <code class="highlighter-rouge">raw.d</code>. This is how we designate the folders containing our raw sequencing data in the Banfield lab standard workflow. Then <code class="highlighter-rouge">cd</code> into that directory. Now let’s fetch your sequence files.</p> <p>The reads are located at <code class="highlighter-rouge">/class_data/reads/</code>. I’ve sent out the full path to your reads for each group on Slack. Because these are large files, we can’t just copy them over to all your home directories- that would instantly use tons of disk space! Instead we’re going to make what are called ‘symbolic links’ of your read files. They look, act and are identical in every way to the original files; luckily, though, they don’t take up any space on the hard drive! They’ll show up as light blue instead of white like other normal files.</p> <p>Here’s how you make a symbolic link to your reads- remember you’ll need to do this for both your forward and reverse reads.</p> <div class="highlighter-rouge"><pre class="highlight"><code>ln -s [EXAMPLE FILENAME] .
</code></pre></div> <p>This creates a link - you can modify the file as normal, but it won’t take up any extra disk space and will let you modify it without providing the full PATH. Otherwise, you’d have to type <code class="highlighter-rouge">/class_data/reads/[SAMPLE NAME]/</code> every time you wanted to access your files. (Which you can still do! I guess. Live your best life.)</p> <p>Here I’ll use an example set of reads. Make sure to use your own set of reads which I provided to your group on Slack. Here’s an example of what to do:</p> <div class="highlighter-rouge"><pre class="highlight"><code>cd ~
mkdir raw.d
cd raw.d
#Remember to use your own reads instead of these example files!
ln -s /class_data/reads/JS_WN1_S130/JS_WN1_S130.1.fastq.gz .
ln -s /class_data/reads/JS_WN1_S130/JS_WN1_S130.2.fastq.gz .
</code></pre></div> <p>Now that you’ve got the data loaded, let’s run our QC program, fastqc! I highly recommend one person per group do this, since it can take quite a while; this way you can get started on the rest of the lab while it’s running. ()</p> <h3 id="running-fastqc">Running FastQC</h3> <p>FastQC is a program to assess the quality of raw sequencing reads. We’re going to use this before and after trimming your reads to highlight the difference trimming makes.</p> <p>FastQC is loaded already onto the cluster, so you all have access. Try running the help menu with <code class="highlighter-rouge">fastqc -h</code>. The output should look like this:</p> <div class="highlighter-rouge"><pre class="highlight"><code>FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

fastqc seqfile1 seqfile2 .. seqfileN

fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam]
[-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

FastQC reads a set of sequence files and produces from each one a quality
control report consisting of a number of different modules, each one of
which will help to identify a different potential type of problem in your
data.

If no files to process are specified on the command line then the program
will start as an interactive graphical application.  If files are provided
on the command line then the program will run with no user interaction
required.  In this mode it is suitable for inclusion into a standardised
analysis pipeline.

The options for the program as as follows:

-h --help       Print this help file and exit

-v --version    Print the version of the program and exit

-o --outdir     Create all output files in the specified output directory.
        Please note that this directory must exist as the program
        will not create it.  If this option is not set then the
        output file for each sequence file is created in the same
        directory as the sequence file which was processed.
</code></pre></div> <p>Read the help menu carefully. FastQC will analyze both your forward and reverse reads at once, so make sure to provide them both!</p> <p>You have the option of putting the output in a directory (by providing a value for the <code class="highlighter-rouge">-o</code> flag), which is tidy, or just having FastQC output everything in your current directory. I recommend making a directory to put the output in, like so:</p> <div class="highlighter-rouge"><pre class="highlight"><code>mkdir fastQC_output
fastqc -o fastQC_output [forward_reads] [reverse_reads]
</code></pre></div> <p>This took about 15 minutes running it on the data I use for my research- I would say you should expect it to take about as long on your samples.</p> <p>Once this is done, you’re going to have an output .html file (either in your current directory or in the output directory you specified with <code class="highlighter-rouge">-o</code>). We’re going to want to download it! This file is an .html file, so you want to open it in a web browser- Firefox, Chrome, Safari, Internet Explorer 6, Netscape, whatever you want.</p> <h3 id="how-to-download-files-from-the-cluster-important">How to Download Files From The Cluster (IMPORTANT)</h3> <p>This is a very important skill, and one that will come up many times throughout the semester. Historically, it’s been a bit tricky, so I want to devote some time to it.</p> <p>You have two options. The easiest of these by far if you’re on Mac or Windows is to use <a href="https://cyberduck.io/">Cyberduck</a>:</p> <ul> <li>Download using a GUI program like <a href="https://winscp.net/eng/download.php">WinSCP</a> for Windows, <a href="https://cyberduck.io/">Cyberduck</a> for Windows or Mac OSX, or Filezilla, I guess, if you’re using Linux. (Instructions on how to use Filezilla for this purpose are <a href="https://stackoverflow.com/questions/299412/is-there-any-winscp-equivalent-for-linux">here</a> although I really recommend you just use SCP if you’re on linux.)</li> </ul> <p>GUI-based SCP applications are pretty straightforward. If you have questions about them ask me in class or on slack.</p> <ul> <li> <p>Download with SCP. Here’s an example of how to do that (I also will give a demonstration in class):</p> </li> <li> <p>Get the full path to the file you want to download and copy it!</p> </li> </ul> <div class="highlighter-rouge"><pre class="highlight"><code>   realpath example_fastqc_output.html
   # OUTPUT: /home/jwestrob/fastqc/example_fastqc_output.html
   # Copy that! ^
</code></pre></div> <ul> <li>Open a terminal on your computer (NOT the class server- you can leave that one open or close it) and navigate to the folder you want to download to.</li> </ul> <div class="highlighter-rouge"><pre class="highlight"><code>  cd ~/Downloads/
  scp jwestrob@class.ggkbase.berkeley.edu:/home/jwestrob/fastqc/example_fastqc_output.html .
  # In Bash, "." just means "here", i.e. your current location.                            
</code></pre></div> <h3 id="trimming-with-sickle">Trimming with Sickle</h3> <p>Next we’re going to use Sickle to trim your reads with <code class="highlighter-rouge">sickle</code>. Try running <code class="highlighter-rouge">sickle pe -h</code> for help (the <code class="highlighter-rouge">pe</code> stands for <code class="highlighter-rouge">paired-end</code>). This yields the following help information:</p> <div class="highlighter-rouge"><pre class="highlight"><code>If you have separate files for forward and reverse reads:
Usage: sickle pe [options] -f &lt;paired-end forward fastq file&gt; -r &lt;paired-end reverse fastq file&gt; -t &lt;quality type&gt; -o &lt;trimmed PE forward file&gt; -p &lt;trimmed PE reverse file&gt; -s &lt;trimmed singles file&gt;

If you have one file with interleaved forward and reverse reads:
Usage: sickle pe [options] -c &lt;interleaved input file&gt; -t &lt;quality type&gt; -m &lt;interleaved trimmed paired-end output&gt; -s &lt;trimmed singles file&gt;

If you have one file with interleaved reads as input and you want ONLY one interleaved file as output:
Usage: sickle pe [options] -c &lt;interleaved input file&gt; -t &lt;quality type&gt; -M &lt;interleaved trimmed output&gt;

Options:
Paired-end separated reads
--------------------------
-f, --pe-file1, Input paired-end forward fastq file (Input files must have same number of records)
-r, --pe-file2, Input paired-end reverse fastq file
-o, --output-pe1, Output trimmed forward fastq file
-p, --output-pe2, Output trimmed reverse fastq file. Must use -s option.

Paired-end interleaved reads
----------------------------
-c, --pe-combo, Combined (interleaved) input paired-end fastq
-m, --output-combo, Output combined (interleaved) paired-end fastq file. Must use -s option.
-M, --output-combo-all, Output combined (interleaved) paired-end fastq file with any discarded read written to output file as a single N. Cannot be used with the -s option.
</code></pre></div> <p>There’s more, but I’ll let you read it on your terminal instead of putting it here.</p> <p>So remember, you have your forward and reverse reads, so you’ll be using the first option-</p> <div class="highlighter-rouge"><pre class="highlight"><code>If you have separate files for forward and reverse reads:
Usage: sickle pe [options] -f &lt;paired-end forward fastq file&gt; -r &lt;paired-end reverse fastq file&gt; -t &lt;quality type&gt; -o &lt;trimmed PE forward file&gt; -p &lt;trimmed PE reverse file&gt; -s &lt;trimmed singles file&gt;
</code></pre></div> <p>Here’s an example, again using the example dataset. Remember not to copy paste this, because your sample name will be different!</p> <div class="highlighter-rouge"><pre class="highlight"><code>sickle pe -t sanger -f JS_WN1_S130.1.fastq.gz -r JS_WN1_S130.2.fastq.gz -o JS_WN1_S130.trimmed.1.fastq.gz -p JS_WN1_S130.trimmed.2.fastq.gz -s JS_WN1_S130.trimmed.single.fastq.gz
</code></pre></div> <p>Notice the flag <code class="highlighter-rouge">-t sanger</code>; this indicates the way the per-base quality scores are formatted. This should take a couple more minutes. Also, make sure the output filenames (<code class="highlighter-rouge">-o</code>, <code class="highlighter-rouge">-p</code>, and <code class="highlighter-rouge">-s</code>) are different than the input filenames, so you don’t accidentally overwrite your files. (If you do, just let me know and I’ll get you the original data back.)</p> <p>This step should take about 5 minutes or so. Don’t worry if it goes longer. (Every sample is a different size.)</p> <h3 id="final-step-fastqc-on-the-trimmed-reads">Final step: FastQC on the trimmed reads</h3> <p>So now that you’ve got trimmed reads, we want to get FastQC results that show how much our quality scores have improved. Run FastQC again, this time using only the trimmed reads.</p> <p>I recommend making another directory for the trimmed FastQC output, like so:</p> <div class="highlighter-rouge"><pre class="highlight"><code>mkdir fastQC_trimmed_output
fastqc -o fastQC_trimmed_output [trimmed_forward_reads] [trimmed_reverse_reads] [trimmed_singleton_reads]
</code></pre></div> <p>Note how I included the singleton reads here (i.e., reads without a mate pair). You don’t have to do this, but I recommend it!</p> <p>Now, as you did before, download the FastQC output and open it on your computer.</p> <h1 id="questions">Questions</h1> <ol> <li> <p>Do you see anything worrying in terms of quality scores in your untrimmed reads?</p> </li> <li> <p>What are the noticeable differences in the FastQC output between your trimmed and untrimmed reads?</p> </li> <li> <p>Briefly describe the pros and cons of short read vs. long read sequencing, and a project that would be appropriate (or inappropriate) to use them for.</p> </li> <li> <p>Based on the average read length and number of reads for one of your samples, answer the following question: What % of the community does a microbe with a 3,000,000 bp genome need to be at in order to be recovered at 10x coverage?</p> </li> </ol> <div class="entry-meta"> <br> <hr> <span class="entry-tags"><a href="/tags/#test" title="Pages tagged test" class="tag"><span class="term">test</span></a></span> <span class="social-share"> <a href="https://www.facebook.com/sharer/sharer.php?u=/Week_2_walkthrough/" title="Share on Facebook" class="tag"> <span class="term"><i class="fa fa-facebook-square"></i> Share</span> </a> <a href="https://twitter.com/intent/tweet?text=/Week_2_walkthrough/" title="Share on Twitter" class="tag"> <span class="term"><i class="fa fa-twitter-square"></i> Tweet</span> </a> <a href="https://plus.google.com/share?url=/Week_2_walkthrough/" title="Share on Google+" class="tag"> <span class="term"><i class="fa fa-google-plus-square"></i> +1</span> </a> </span> <div style="clear:both"></div> </div> </div> </div> </header> <!-- JS --> <script src="/assets/js/jquery-1.12.0.min.js"></script> <script src="/assets/js/jquery.dlmenu.min.js"></script> <script src="/assets/js/jquery.goup.min.js"></script> <script src="/assets/js/jquery.magnific-popup.min.js"></script> <script src="/assets/js/jquery.fitvid.min.js"></script> <script src="/assets/js/scripts.js"></script> <!-- MathJax --> <script async src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script> </body> </html>
